src/walker/walker.ts

```diff
@@ -275,7 +275,14 @@ export class Walker extends EventEmitter<WalkerEmitter> {
         return [new ReferenceNode(fragment, '$ref is not a string'), fragment];
       } else if (walkingOptions.resolveRef !== null) {
         try {
-          fragment = walkingOptions.resolveRef(path, fragment.$ref);
+          let newFragment = walkingOptions.resolveRef(path, fragment.$ref);
+
+          if (typeof fragment.description === 'string') {
```

Comment:
This is in line with https://github.com/stoplightio/json/blob/master/src/resolveInlineRef.ts#L55.
`summary` is not taken into account, because JST does not consume it anywhere

Reply:
What's the right thing to do though?  Is it ok that resolveInLineRef clobbers description and summary like that?

Reply:
Yeah, that's what we generally do. If there's a `description` next to `$ref`, it should always override it.
Whether it's placed before or after $ref, it doesn't matter.